services/QuillLMS/app/controllers/api/v1/teacher_notification_settings_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class Api::V1::TeacherNotificationSettingsController < ApplicationController
+  before_action :set_current_settings, only: [:index, :bulk_update]
+
+  def index
+    render json: types_and_values
+  end
+
+  def bulk_update
+    TeacherNotificationSetting.transaction do
+      teacher_notification_setting_params.each do |notification_type, active|
+        # Convert the string to a boolean so that 'false' is treated as false
```

Comment:
It might make sense to use a traditional if/else block since the guard clauses have overlap:

```
# Convert the string to a boolean so that 'false' is treated as false
if ActiveModel::Type::Boolean.new.cast(active)
  next if @current_settings.exists?(notification_type: notification_type)

  TeacherNotificationSetting.create!(user: current_user, notification_type: notification_type)
else
  # Destroy settings that are not in the payload params
  @current_settings.find_by(notification_type: notification_type)&.destroy!
end
```